### PR TITLE
Use pyspark 3.0 API rdd.collectWithJobGroup to address cancelling job issue

### PR DESCRIPTION
**Pull request by WeichenXu123**
* Use pyspark 3.0 API rdd.collectWithJobGroup to address cancelling job issue.
* Update CI only test against spark 3.0
* Update README requirements.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This library provides Apache Spark backend for joblib to distribute tasks on a S
 
 ## Installation
 
-`joblibspark` requires Python 3.6+, `joblib>=0.14` and `pyspark>=2.4.4` to run.
+`joblibspark` requires Python 3.6+, `joblib>=0.14` and `pyspark>=3.0.0` to run.
```
Comment (liangz1) [inline README.md:7]: Shall we mention `pyspark==3.0.0` is not available on pypi yet?

File: joblibspark/backend.py [hunk 2]
```diff
@@ -21,20 +21,23 @@
...
 from pyspark import cloudpickle
+from pyspark.util import VersionUtils
 
 
 def register():
     """
     Register joblib spark backend.
     """
+    if VersionUtils.majorMinorVersion(pyspark.__version__)[0] < 3:
+        raise RuntimeError("Joblib-spark only support spark >= 3.0")
```
Comment (liangz1) [inline joblibspark/backend.py:40]: Shall we keep the old code path for spark 2.x users (if any)?

**Comment by WeichenXu123** [comment]
@liangz1 @mengxr  Ready.

File: .travis.yml [hunk 3]
```diff
@@ -14,19 +14,17 @@ python:
   - "3.7"
 
 jdk:
-   - openjdk8
+   - oraclejdk8
```
Comment (mengxr) [inline .travis.yml:17]: Why switching to oracle JDK?

File: .travis.yml [hunk 4]
```diff
@@ -14,19 +14,17 @@ python:
...
 
 matrix:
   include:
-    - env: PYSPARK_PIN_THREAD=true SKLEARN_VERSION=0.21.3 SPARK_VERSION=3.0.0-SNAPSHOT
-    - env: SKLEARN_VERSION=0.21.3 SPARK_VERSION=2.4.4
-
-before_install:
-  - ./bin/download_travis_dependencies.sh
+    - env: PIP_PYSPARK=https://ml-team-public-read.s3-us-west-2.amazonaws.com/pyspark-3.0.1.dev0-860849-20200506.tar.gz
+    - env: PIP_PYSPARK="pyspark==2.4.5" JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk-amd64"
```
Comment (mengxr) [inline .travis.yml:22]: and then use OpenJDK here?

File: joblibspark/backend.py [hunk 5]
```diff
@@ -66,21 +67,14 @@ def __init__(self, **backend_args):
...
-        if self._check_pyspark_pin_thread_mode():
-            self._spark.sparkContext.cancelJobGroup(self._job_group)
-        else:
+        if VersionUtils.majorMinorVersion(pyspark.__version__)[0] < 3:
             # Note: There's bug existing in `sparkContext.cancelJobGroup`.
-            # See https://github.com/apache/spark/pull/24898
-            warnings.warn("Because pyspark py4j is not in pinned thread mode, "
-                          "we could not terminate running spark jobs correctly.")
+            # See https://issues.apache.org/jira/browse/SPARK-31549
+            warnings.warn("For spark version < 3, pyspark cancelling job API has bugs, "
```
Comment (mengxr) [inline joblibspark/backend.py:74]: Shall we put the JIRA link in the error message?

File: README.md (hunk 1 above)
Comment (mengxr) [inline README.md:7]: Need to keep the old one.

File: .travis.yml [hunk 6]
```diff
@@ -18,15 +18,13 @@ jdk:
...
```
Comment (mengxr) [inline .travis.yml:22]: Do we need to set JAVA_HOME?

Comment (WeichenXu123) [reply .travis.yml:22]: Yes. Travis xenial use java11 which is incompatible with spark 2.4.

**Comment by mengxr** [comment]
LGTM

**Pull request merged**